src/core/raster/qgsrasterpipe.h

```diff
@@ -73,6 +73,21 @@ class CORE_EXPORT QgsRasterPipe
 
     QgsRasterPipe &operator=( const QgsRasterPipe &rh ) = delete;
 
+    /**
+     * Moves the pipe to another \a thread.
+     *
+     * This effects all QObject derived interfaces in the pipe, and follows the same
+     * behavior as QObject::moveToThread. Specifically, it is permitted to PUSH the
+     * pipe from the current thread to another thread, but NOT to PULL a pipe
+     * from another thread over to the current thread. Pulling is only supported
+     * by first pushsing the pipe from its current thread to a NULLPTR thread,
+     * and then later pulling to the current thread. See QObject documentation
+     * for more details.
+     *
+     * \since QGIS 3.30
+    */
+    void moveToThread( QThread *thread );
```

Comment:
We're OK exposing that to the python API?

Reply:
yeah, might be needed for custom processing algorithms.

Reply:
K